File: bms/go/go-edlib/union/union.go
package test

var ()

const ()

func union(a, b []string) [][]rune {
	m := make(map[string]bool)
	for _, item := range a {
		m[item] = true
	}
	for _, item := range b {
		if _, ok := m[item]; !ok {
			a = append(a, item)
		}
	}

	// Convert a to rune matrix (with x -> words and y -> characters)
	out := make([][]rune, len(a))
	for i, word := range a {
		out[i] = []rune(word)
	}
	return out
}

File: translations_only/fail/go/go-edlib/union/gemini/CAPR/union.rs

use std::collections::HashMap;

pub fn union(mut a: Vec<String>, b: Vec<String>) -> Vec<Vec<char>> {
    let mut m: HashMap<String, bool> = HashMap::new();
    for item in a.iter() {
        m.insert(item.clone(), true);
    }
    for item in b.iter() {
        if !m.contains_key(item) {
            a.push(item.clone());
        }
    }

    // Convert a to rune matrix (with x -> words and y -> characters)
    let mut out: Vec<Vec<char>> = Vec::new();
    for word in a.iter() {
        out.push(word.chars().collect());
    }
    out
}
